src/main/kotlin/org/scoverage/ScoverageAggregate.kt

```diff
@@ -0,0 +1,74 @@
+package org.scoverage
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileCollection
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.TaskAction
+import scoverage.report.CoverageAggregator
+
+import org.gradle.api.tasks.PathSensitivity.RELATIVE
+import java.io.File
+
+abstract class ScoverageAggregate: DefaultTask() {
```

Comment:
Note that all properties here, including the `@Nested` one, can be defined as `abstract val`s, removing the need to call `objects` methods explicitly. 

This is called "managed properties", and is described [here](https://docs.gradle.org/current/userguide/custom_gradle_types.html). It is considered idiomatic to declare as many properties as possible as managed.